proto/core.rst

```diff
@@ -1,25 +1,403 @@
+| :param:`importmap`  | :type:`string`       | :value:`""`                                     |
++---------------------+----------------------+-------------------------------------------------+
+| The Go package path of this library. This is mostly only visible to the                      |
+| compiler and linker, but it may also be seen in stack traces. This may be                    |
+| set to prevent a binary from linking multiple packages with the same import                  |
+| path, e.g., from different vendor directories.                                               |
++---------------------+----------------------+-------------------------------------------------+
+| :param:`embed`      | :type:`label_list`   | :value:`[]`                                     |
++---------------------+----------------------+-------------------------------------------------+
+| List of Go libraries that should be combined with this library. The ``srcs``                 |
+| and ``deps`` from these libraries will be incorporated this library when it                  |
+| is compiled. Embedded libraries must have the same ``importpath`` and                        |
+| Go package name.                                                                             |
++---------------------+----------------------+-------------------------------------------------+
+| :param:`gc_goopts`  | :type:`string_list`  | :value:`[]`                                     |
++---------------------+----------------------+-------------------------------------------------+
+| List of flags to add to the Go compilation command when using the gc                         |
+| compiler. Subject to `Make variable substitution`_ and `Bourne shell tokenization`_.         |
++---------------------+----------------------+-------------------------------------------------+
+| :param:`compiler`   | :type:`label`        | :value:`None`                                   |
++---------------------+----------------------+-------------------------------------------------+
+| Equivalent to ``compilers`` with a single label.                                             |
++---------------------+----------------------+-------------------------------------------------+
+| :param:`compilers`  | :type:`label_list`   | :value:`["@io_bazel_rules_go//proto:go_proto"]` |
++---------------------+----------------------+-------------------------------------------------+
+| List of rules producing `GoProtoCompiler`_ providers (normally                               |
+| `go_proto_compiler`_ rules). This is usually understood to be a list of                      |
+| protoc plugins used to generate Go code. See `Predefined plugins`_ for                       |
+| some options.                                                                                |
++---------------------+----------------------+-------------------------------------------------+
+
+Example: Basic proto
```

Comment:
At some point we should think about the cleanest way to have working examples that are easy to link to from the documentation.